spec.html

```diff
@@ -8362,6 +8351,17 @@ <h1>ECMAScript Function Objects</h1>
             The <emu-xref href="#sec-source-text">source text</emu-xref> that defines the function.
           </td>
         </tr>
+        <tr>
+          <td>
+            [[CallThrowsTypeError]]
```

Comment:
I agree @zenparsing's original comment here. `[[CallThrowsTypeError]]` reads oddly, is too specific, and doesn't impart much understanding. How about `[[MustConstruct]]`?

Reply:
I agree it reads oddly if you start out thinking that "Call" is a verb. But once you realize it's a reference to `[[Call]]`, and should be read as a noun, it reads okay (in my opinion).

As for "too specific", it's exactly as specific as its only effect. If it had some additional effect, I would agree it needs a more general name.

But anyhow, if the editors agree on a different name, I'll change it.

Reply:
> I agree it reads oddly if you start out thinking that "Call" is a verb. But once you realize it's a reference to [[Call]], and should be read as a noun, it reads okay (in my opinion).

Indeed, and I don't think that kind of realization should be needed. @ljharb, could you confirm you're fine with `[[MustConstruct]]`? If so, please change it.

With both this and below renamed, I reckon it's good to land.

Reply:
Yes, "MustConstruct" conveys the same information as "Call will throw" imo :-)

Reply:
Another possibility would be `[[IsClassConstructor]]`, since `MakeClassConstructor` is the only thing that would set it to `*true*`.

(We have the chain of implications:
- It's a class constructor. ->
- It can only be invoked as a constructor. ->
- An attempt to invoke it as a function must fail. ->
- Its `[[Call]]` method must throw a `*TypeError*`.

and we're picking different points on this chain to name the slot. If the last point doesn't impart much understanding, maybe the first point imparts the most?)